pkg/lib/authn/authenticator/whatsapp/store_redis.go

```diff
@@ -0,0 +1,98 @@
+package whatsapp
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+
+	goredis "github.com/go-redis/redis/v8"
+
+	"github.com/authgear/authgear-server/pkg/lib/config"
+	"github.com/authgear/authgear-server/pkg/lib/infra/redis/appredis"
+	"github.com/authgear/authgear-server/pkg/util/clock"
+)
+
+type StoreRedis struct {
+	Redis *appredis.Handle
+	AppID config.AppID
+	Clock clock.Clock
+}
+
+func (s *StoreRedis) Create(code *Code) error {
+	ctx := context.Background()
```

Comment:
It's better to inject the request context here, so that redis operation can have the same deadline of the request.